### PR TITLE
doc: release 3.6: no CACHE_MANAGEMENT by default on stm32h7/f7

### LINKED ISSUE #75487: Program execution takes more time in zephyr 3.6.0 on stm32-h743 controller
**PROBLEM**
Program execution takes more time in zephyr 3.6.0 on stm32-h743 controller

**Elaboration:**
Same program execution takes more time in zephyr 3.6.0 when compared with zephyr 3.2.0 on stm32-h743 controller with same default system clock (96MHZ) whereas on stm32-u573 it works fine.

**To Reproduce**
-Remove blinky code in main.c at:zephyr\samples\basic\blinky\src and write the code as written below:
	
	volatile uint32_t current_time_uSec1 = 0;
	volatile uint32_t current_time_uSec2 = 0;
	
		while ( 1 )
	{
		current_time_uSec1 = k_cyc_to_us_near32( k_cycle_get_32() );
		for ( volatile uint32_t i = 0; i < 48; i++ )
		{}
		current_time_uSec2 = k_cyc_to_us_near32( k_cycle_get_32() );

		printk("time diff is %d\n", current_time_uSec2 - current_time_uSec1);

		k_msleep( 1000 );
	}

**Build code using:** west build -b nucleo_h743zi ./ --pristine

**OBSERVATION:**
1.Same code on stm32-h743 controller with **zephyr3.6.0 [attached-H7_3_6.PNG] print is coming as 13usec** while with **zephyr 3.2.0 its 4usec** [attached-H7_3_2.PNG] default system clock (96MHZ).
**WHY SO MUCH DIFFERENCE 13usec and 4usec??**
2.**Same code on stm32-u575 controller with zephyr3.6.0 print is coming as 3 usec while with zephyr 3.2.0 its 3 usec.default system clock (160MHZ).**

**Expected behavior**
Same code should not take so much of execution time between two zephyr versions.

**Impact**
Performance.

**Logs and console output**
H7_3_6
H7_3_2

**Environment (please complete the following information):**
zephyr-sdk-0.16.5-1 with zephyr 3.6.0
zephyr-sdk-0.15.2 with zephyr 3.2.0

### PR DESCRIPTION
**Pull request by FRASTM**
Mentioning that stm32h7 and stm32F7 series do not have the CONFIG_CACHE_MANAGEMENT by default. The application must explicitely set CONFIG_CACHE_MANAGEMENT=y to activate the cache (Icache, Dcache) on those stm32 series.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/75487

### CODE REVIEW

**Comment by FRASTM** [comment]
Not sure about the doc/release and  milestone for that PR

**Comment by henrikbrixandersen** [comment]
We usually do not change release notes for past releases.

**Comment by erwango** [comment]
> We usually do not change release notes for past releases.

Agreed, but not documenting it was actually an issue.

Anyone migrating from V3.5 to any other release after that will face this issue.
This is the only place where makes sense to put the update as they will (should at least) read this migration guide.

**Review by aescolar — DISMISSED** [review]
I'm ok with this, but @kartben please comment

**Comment by aescolar** [comment]
I guess you can also set it to be backported to 3.6 :)

File: doc/releases/migration-guide-3.6.rst [hunk 1]
```diff
@@ -46,6 +46,9 @@ Kernel
   :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` option has been introduced (which defaults
   being disabled).
 
+* STM32H7 and STM32F7 can activate the cache (Icache and Dcache) by setting explicitly
+  the  ``CONFIG_CACHE_MANAGEMENT`` to ``y``.
```
Comment (erwango) [inline doc/releases/migration-guide-3.6.rst:50]:
```suggestion
* STM32H7 and STM32F7 should now activate the cache (Icache and Dcache) by setting explicitly
  the  ``CONFIG_CACHE_MANAGEMENT`` to ``y``.
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
> I guess you can also set it to be backported to 3.6 :)

Indeed. Also requested to be backported on 3.7 (as to be seen by anyone coming from pre-3.6 directly to 3.7)

File: doc/releases/migration-guide-3.6.rst (hunk 1 above)
Comment (FRASTM) [reply doc/releases/migration-guide-3.6.rst]: done

**Review by erwango — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Comment by aescolar** [comment]
@kartben can you please tell if you think this is ok

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**